web/lib/HTTPError.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @class
+ * @param {Number}          status  HTTP Status Code
+ * @param {Error|String}    err     Original Error or Human Readable message if no error is present
+ * @param {String}          human   Human Readable Error Message
+ *
+ * @prop {Number}           status  HTTP Status Code
+ * @prop {Error}            err     Original Error Object
+ * @prop {String}           human   Human readable string
+ */
+class HTTPError {
+    constructor(status, err, human) {
+        console.error(err);
+
+        this.status = status;
+
+        if (err instanceof Error) {
+            this.err = err;
+        }
+
+        if (typeof err === 'string') {
+            this.human = err;
+        } else if (!human) {
+            this.human = 'Unknown Error';
+        } else {
+            this.human = err;
+        }
+    }
+
+    respond(res) {
+        res.status(this.status).json({
+            status: this.status,
+            message: this.human
+        });
+    }
+}
+
+/**
+ * Standard error logging and internal error response
+ *
+ * @param {Object} err HTTPError or Error like object
+ * @param {Object} req Express Request Object
+ * @param {Object} res Express Result Object
+ */
+function HTTPErrorMiddleware(err, req, res) {
+    if (err instanceof HTTPError) return err.respond(res);
+
+    if (err.code === 'SQLITE_CONSTRAINT') {
```

Comment:
The return logic is sometimes defined here, sometimes in ``HTTPError::respond``. Can it be unified so we're always running through that method?